src/cargo/util/toml/targets.rs

```diff
@@ -30,7 +30,7 @@ use crate::util::toml::warn_on_deprecated;
 const DEFAULT_TEST_DIR_NAME: &'static str = "tests";
 const DEFAULT_BENCH_DIR_NAME: &'static str = "benches";
 const DEFAULT_EXAMPLE_DIR_NAME: &'static str = "examples";
-const DEFAULT_BIN_DIR_NAME: &'static str = "bin";
+const DEFAULT_BIN_DIR_NAME: &'static str = "src/bin";
```

Comment:
> Is this safe on Windows if pkg_root is accidentally canonicalized and becomes \\?\?

So much other stuff will break in that situation iiuc.  For example, we do the same thing for all explicit paths in a manifest.

> I understand that in Cargo we use paths::normalize_path to avoid canonicalization, though I am not sure how things will evolve.

We'd have a lot to fix up if we want to go that direction.  For myself, I found doing this cleaned up the code.  So we'd need to weigh out the cleaner code against making it slightly more work in the future *if* we switch to canonicalized paths.